### PR TITLE
Fix bug 1390111 Add migration to unescape quotes from android_strings…

### PR DESCRIPTION
**Pull request by jotes**
Hey @mathjazz,

As we talked at irc -> could you look if that migration is acceptable?

Thx!

### CODE REVIEW

File: pontoon/base/migrations/0105_unescape_quotes_from_android_strings_dtd_in_translation_memory.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+from django.db import migrations
+from django.db.models import Q, F, Func, Value
+
+
+def unescape_quotes_in_tm_entries_from_android_strings_dtd(apps, schema):
+    """
+    Problem with the synchronization of android_strings.dtd (&quot; and &apos; entities) was fixed in eb5f8d553.
+    However, TM still contains entries with broken entities which break TMX files.
+
+    Related bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1390111
```
Comment (mathjazz) [inline pontoon/base/migrations/0105_unescape_quotes_from_android_strings_dtd_in_translation_memory.py]: I'd also mention https://github.com/mozilla/pontoon/commit/4537eac40f36cff404fc36bc9f7a5c4c82b520e8 and bug 1390113. Let's change this comment to:

```
Handling of different escape sequences for mobile Android strings was fixed in eb5f8d5. However,
data migration in 4537eac only covered the Translation model, leaving out TranslationMemoryEntry.
See bugs 1390111 and 1390113 for more details.
```

File: pontoon/base/migrations/0105_unescape_quotes_from_android_strings_dtd_in_translation_memory.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+        def replace(expr, substring, replacement):
+            """Shortcut for SQL Replace function."""
+            return Func(expr, Value(substring), Value(replacement), function='replace')
+
+        return replace(
+            replace(F(fieldname), '\\u0022', '"'), '\\u0027', "'"
+        )
+
+    TranslationMemoryEntry.objects.filter(
```
Comment (mathjazz) [inline pontoon/base/migrations/0105_unescape_quotes_from_android_strings_dtd_in_translation_memory.py]: We should limit the query to `entity__resource__path__contains="mobile/android/base"`, as we did in https://github.com/mozilla/pontoon/commit/4537eac40f36cff404fc36bc9f7a5c4c82b520e8.

File: pontoon/base/migrations/0105_unescape_quotes_from_android_strings_dtd_in_translation_memory.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    def unescape_quotes(fieldname):
+        """Get a SQL function which will unescape quotes in a fieldname."""
+
+        def replace(expr, substring, replacement):
+            """Shortcut for SQL Replace function."""
+            return Func(expr, Value(substring), Value(replacement), function='replace')
+
+        return replace(
+            replace(F(fieldname), '\\u0022', '"'), '\\u0027', "'"
+        )
```
Comment (mathjazz) [inline pontoon/base/migrations/0105_unescape_quotes_from_android_strings_dtd_in_translation_memory.py]: I find this function very hard to read. I suggest we use 4 DB queries instead:

```python
TranslationMemoryEntries = TranslationMemoryEntry.objects.filter(
    entity__resource__path__contains="mobile/android/base"
)

TranslationMemoryEntries.filter(target__contains='\\u0022').update(
    target=Func(
        F('target'),
        Value('\\u0022'), Value('"'),
        function='replace',
    )
)

TranslationMemoryEntries.filter(target__contains='\\u0027').update(
    target=Func(
        F('target'),
        Value('\\u0027'), Value("'"),
        function='replace',
    )
)

TranslationMemoryEntries.filter(source__contains='\\u0022').update(
    source=Func(
        F('source'),
        Value('\\u0022'), Value('"'),
        function='replace',
    )
)

TranslationMemoryEntries.filter(source__contains='\\u0027').update(
    source=Func(
        F('source'),
        Value('\\u0027'), Value("'"),
        function='replace',
    )
)
```

**Comment by jotes** [comment]
@mathjazz fixed :-)

**Pull request merged**